common/djangoapps/util/enterprise_helpers.py

```diff
@@ -0,0 +1,125 @@
+    )
+
+except ImportError:
+    pass
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+
+def enterprise_enabled():
+    """
+    Determines whether the Enterprise app is installed
+    """
+    return 'enterprise' in settings.INSTALLED_APPS
+
+
+def data_sharing_consent_requested(request):
+    """
+    Determine if the EnterpriseCustomer for a given HTTP request
+    requests data sharing consent
+    """
+    if enterprise_enabled():
+        return active_provider_requests_data_sharing(request)
+    else:
+        return False
+
+
+def data_sharing_consent_required_at_login(request):
+    """
+    Determines if data sharing consent is required at
+    a given location
+    """
+    if enterprise_enabled():
+        return active_provider_enforces_data_sharing(request, EnterpriseCustomer.AT_LOGIN)
+    else:
+        return False
+
+
+def data_sharing_consent_requirement_at_login(request):
+    """
+    Returns either 'optional' or 'required' based on where we are.
+    """
+    if not enterprise_enabled():
+        return None
+    if data_sharing_consent_required_at_login(request):
+        return 'required'
+    if data_sharing_consent_requested(request):
+        return 'optional'
+    return None
+
+
+def insert_enterprise_fields(request, form_desc):
+    """
+    For right now, a simple wrapper for `add_data_sharing_consent_field`,
+    but we might want to add other form fields later.
+    """
+    if not enterprise_enabled():
+        return
+    add_data_sharing_consent_field(request, form_desc)
+
+
+def add_data_sharing_consent_field(request, form_desc):
+    """
+    Adds a checkbox field to be selected if the user consents to share data with
+    the EnterpriseCustomer attached to the SSO provider with which they're authenticating.
+    """
+    enterprise_customer = get_enterprise_customer_for_request(request)
+    required = data_sharing_consent_required_at_login(request)
+
+    if not data_sharing_consent_requested(request):
+        return
+
+    label = _(
+        "I agree to allow {platform_name} to share data about my enrollment, "
+        "completion and performance in all {platform_name} courses and programs "
+        "where my enrollment is sponsored by {ec_name}."
+    ).format(
+        platform_name=configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME),
+        ec_name=enterprise_customer.name
+    )
+
+    error_msg = _(
+        "To link your account with {ec_name}, you are required to consent to data sharing."
+    ).format(
+        ec_name=enterprise_customer.name
+    )
+
+    form_desc.add_field(
+        "data_sharing_consent",
+        label=label,
+        field_type="checkbox",
+        default=False,
+        required=required,
+        error_messages={"required": error_msg},
+    )
+
+
+def insert_enterprise_pipeline_elements(pipeline):
+    """
+    If the enterprise app is installed, insert additional elements into the
```

Comment:
We actually don't know/care if the enterprise app is installed here -- only that "enterprise" is enabled.